crates/sargon/src/signing/tx_to_sign.rs

```diff
@@ -1,20 +1,28 @@
 use crate::prelude::*;
 
-#[derive(Clone, Debug, PartialEq, Eq, std::hash::Hash)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub(crate) struct TXToSign {
-    pub(crate) intent_hash: IntentHash,
+    pub(crate) intent: TransactionIntent,
     entities_requiring_auth: Vec<AccountOrPersona>, // should be a set but Sets are not `Hash`.
 }
 
+impl Identifiable for TXToSign {
+    type ID = IntentHash;
+
+    fn id(&self) -> Self::ID {
+        self.intent.intent_hash()
```

Comment:
minor optimization is to eagerly calc the `intent_hash` inside the ctor, and keep it as a private field, so that we need not calculate it multiple times like we do here now.

Reply:
Right was thinking about it. 💯